src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -239,10 +244,42 @@ class ReferralService(
     submitAdditionalRiskInformation(referral, user)
     communityAPIReferralService.send(referral)
 
-    val sentReferral = referralRepository.save(referral)
-    eventPublisher.referralSentEvent(sentReferral)
+    referralRepository.save(referral)
+    eventPublisher.referralSentEvent(referral)
```

Comment:
I don't remember why but I think there was a reason for saving the result of the `save` and passing that through. Can we use that approach here too?

Reply:
using the saved variable is passing it through is not working  in the new approach as when we try to set ``` referral.complexityLevelIds = draftReferral.complexityLevelIds```, it was throwing Found shared references to a ElementCollection exception.

Reply:
Resolved.